rdt/hyper_transformer.py

```diff
@@ -653,25 +548,25 @@ def _fit_field_transformer(self, data, field, transformer):
         """
         if transformer is None:
             self._add_field_to_set(field, self._fitted_fields)
-            self._transformers_tree[field]['transformer'] = None
-            self._transformers_tree[field]['outputs'] = [field]
+            if field not in self._output_columns:
+                self._output_columns.append(field)
 
         else:
             transformer = get_transformer_instance(transformer)
             transformer.fit(data, field)
-            self._add_field_to_set(field, self._fitted_fields)
             self._transformers_sequence.append(transformer)
             data = transformer.transform(data)
 
-            output_sdtypes = transformer.get_output_sdtypes()
+            output_columns = transformer.get_output_columns()
             next_transformers = transformer.get_next_transformers()
-            self._transformers_tree[field]['transformer'] = transformer
-            self._transformers_tree[field]['outputs'] = list(output_sdtypes)
-            for output_name in output_sdtypes:
+            for output_name in output_columns:
                 output_field = self._multi_column_fields.get(output_name, output_name)
                 next_transformer = next_transformers[output_field]
-                if next_transformer and self._field_in_data(output_field, data):
-                    self._fit_field_transformer(data, output_field, next_transformer)
+
+                # If the column is part of a multi-column field, and at least one column
+                # isn't present in the data, then it should not fit the next transformer
+                if self._field_in_data(output_field, data):
+                    data = self._fit_field_transformer(data, output_field, next_transformer)
 
         return data
```

Comment:
@pvk-developer @amontanez24 This method changes the data inplace, so there's no need to return the data. I ran into a problem with the mocks when trying to make this change though, so I ended up undoing it instead, since I didn't care enough to bother figuring out what the problem is.

If either of you thinks this would indeed be an improvement, let me know and I'll fix it, otherwise I'll just leave it as is.